relay-common/src/utils.rs

```diff
@@ -1,11 +1,49 @@
 use std::fmt;
+use std::mem;
 use std::str;
+use std::sync::{Arc, Mutex};
 
 use once_cell::sync::OnceCell;
 use regex::Regex;
 
 use crate::macros::impl_str_serde;
 
+/// Wraps either the provided pattern string, which will be consumed and replaced with compiled
+/// regex, when the `compiled` function is called.
+#[derive(Debug, Clone)]
+struct GlobRegex {
```

Comment:
I'm realizing that we also have the `Pattern` type in Relay that's used in the PiiConfig that deserializes eagerly:

https://github.com/getsentry/relay/blob/35d2c744a141acc917da8904bec5c5eb4f5bf4cb/relay-general/src/pii/config.rs#L20-L59

Do you see a straight-forward way of covering both in this PR?

Reply:
I will look into this.